src/library/scala/collection/immutable/Queue.scala

```diff
@@ -139,7 +141,10 @@ sealed class Queue[+A] protected(protected val in: List[A], protected val out: L
     *
     *  @param  iter        an iterable object
     */
-  def enqueueAll[B >: A](iter: scala.collection.Iterable[B]) = new Queue(iter.toList reverse_::: in, out)
+  def enqueueAll[B >: A](iter: scala.collection.Iterable[B]) =
+    if (isEmpty) new Queue(in, iter.toList ::: out)
```

Comment:
if it's empty, `in` and `out` are both `Nil`, so I think it could slightly simpler as:
```suggestion
    if (isEmpty) new Queue(Nil, iter.toList)
```

Reply:
Hi, thank you for your message. I will give a look again and make the suggested changes. Let me know if there is anything else I can do to conclude this ticket.